### PR TITLE
Feat: `POST /api/wallet/private_key` API

### PR DESCRIPTION
**Pull request by OjusWiZard**
## Proposed changes

Adds `POST /api/wallet/private_key` to retrieve the private key using the provided password for the given ledger type.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+from fastapi.testclient import TestClient
+
+from operate.cli import create_app
+from operate.operate_types import LedgerType
+
+
+@pytest.fixture
+def mock_password() -> str:
+    """Fixture to provide a mock password for testing."""
+    return "test_password"
```
Comment (jmoreira-valory) [inline tests/test_api.py:36]: There is a common function on the fixtures file for this.

File: tests/test_api.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    return client
+
+
+@pytest.mark.parametrize("logged_in", [True, False])
+@pytest.mark.parametrize(
+    "case",
+    [
+        ("test_password", True),
+        ("wrong_password", False),
+    ],
```
Comment (jmoreira-valory) [inline tests/test_api.py:72]: Also test for None, etc.
